src/components/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts

```diff
@@ -75,6 +121,37 @@ export const getPipelineFromPipelineRun = (pipelineRun: PipelineRunKind): Pipeli
   };
 };
 
+export const createStepStatus = (
+  stepName: string,
+  status: PLRTaskRunDataStatus & { reason: RunStatus; duration: string },
+): StepStatus => {
+  let stepRunStatus: RunStatus = RunStatus.Pending;
+  let duration: string = null;
+
+  const matchingStep: PLRTaskRunStep = getMatchingStep(stepName, status);
+  if (!status || !status.reason) {
+    stepRunStatus = RunStatus.Cancelled;
+  } else {
+    if (!matchingStep) {
+      stepRunStatus = RunStatus.Pending;
+    } else if (matchingStep.terminated) {
+      stepRunStatus = terminatedReasonToRunStatus(matchingStep.terminated.reason);
+      duration = getStepDuration(matchingStep) || status.duration;
+    } else if (matchingStep.running) {
+      stepRunStatus = RunStatus['In Progress'];
```

Comment:
The RunStatus that comes from pipelines topology don't have a gap in-between, either remove the gap or change it to `RunStatus.Running`